Solutions/Azure Activity/Analytic Rules/RareOperations.yaml

```diff
@@ -23,32 +21,36 @@ relevantTechniques:
 query: |
   let starttime = 14d;
   let endtime = 1d;
-  // The number of operations below which an IP address is considered an unusual source of role assignment operations
+  // The number of operations above which an IP address is considered an unusual source of role assignment operations
   let alertOperationThreshold = 5;
+  // Add or remove operation names below as per your requirements. For operations lists, please refer to https://learn.microsoft.com/en-us/Azure/role-based-access-control/resource-provider-operations#all
   let SensitiveOperationList =  dynamic(["microsoft.compute/snapshots/write", "microsoft.network/networksecuritygroups/write", "microsoft.storage/storageaccounts/listkeys/action"]);
   let SensitiveActivity = AzureActivity
   | where OperationNameValue in~ (SensitiveOperationList) or OperationNameValue hassuffix "listkeys/action"
   | where ActivityStatusValue =~ "Success";
   SensitiveActivity
   | where TimeGenerated between (ago(starttime) .. ago(endtime))
-  | summarize count() by CallerIpAddress, Caller, OperationNameValue
+  | summarize count() by CallerIpAddress, Caller, OperationNameValue, bin(TimeGenerated,1d)
   | where count_ >= alertOperationThreshold
```

Comment:
We've found using series_decompose_anomalies to be much more effective than thresholding for things like this.

Reply:
I haven't changed the original logic in this case. Static threshold with anomaly was being used in this rule. I think it would require additional testing if we alter the logic. Kept it the same.